antarest/study/service.py

```diff
@@ -1661,7 +1661,7 @@ def set_public_mode(self, study_id: str, mode: PublicMode) -> None:
     def check_errors(self, uuid: str) -> List[str]:
         study = self.get_study(uuid)
         self.assert_study_unarchived(study)
-        return self.storage_service.raw_study_service.check_errors(study)
+        return self.storage_service.raw_study_service.check_errors(cast(RawStudy, study))
```

Comment:
This code is actually currently buggy:
it can be called for a variant study through the API, whereas it should not.

I think we should just remove it, since it's actually not used and not maintained. (in that case we must document it as a breaking change in release notes)

If we keep it, here we should raise an exception if it's not a RawStudy